src/layer.rs

```diff
@@ -597,6 +638,15 @@ where
     /// # drop(subscriber);
     /// ```
     pub fn new(tracer: T) -> Self {
+        let tracer = Arc::new(tracer);
+
+        let build_unbuilt_span = {
```

Comment:
This can't be a function pointer, because it needs to capture the tracer Arc, which is why this looks different to `Self::get_context`